src/Language/Haskell/Liquid/GHC/Plugin.hs

```diff
@@ -285,12 +294,12 @@ processInputSpec cfg pipelineData modSummary tcGblEnv inputSpec = do
       }
 
   -- liftIO $ putStrLn ("liquidHaskellCheck 6: " ++ show isIg)
-  liquidLib' :: Either LiquidCheckException LiquidLib <-
+  liquidLib' :: Either LiquidCheckException (LiquidLib, [Filter]) <-
     if isIgnore inputSpec
-    then pure $ Right emptyLiquidLib
+    then pure $ Left KillPluginWithSuccess
     else checkLiquidHaskellContext lhContext
 
-  traverse (serialiseSpec thisModule tcGblEnv) liquidLib'
+  traverse (bitraverse (serialiseSpec thisModule tcGblEnv) pure) liquidLib'
```

Comment:
Apparently this is evaluated, even if the earlier call to checkLiquidHaskellContext returns KillPluginWithSuccess.

Reply:
We agreed during the review that this code is correct, but non-obvious, and needs to be changed to make the control flow explicit.